Updated contact tracing flow for location history

davidmiller opened the pull request:
Preview of inpatient contact tracing flow (Location history, contacts, bed history, slice detail)

The instances of scrollable-table-container likely need replacing with some position fixed +fixed height options at varying heights

Requires a fully sync'd IPC status, location history, bed_status to preview

Commit:
Changes ipc.loader.load_all_extra_ipc_patients to look at merged mrns

Changes the load_all_extra_ipc_patients to also check the merged MRN
table before creating patients.

Commit:
Changes the IPC episode loader to work with merged MRNs

Previously we assumed we could bridge bed status and infection control
by the value provided in the upstream MRN.

Now that we are accounting for patient merges this is no longer True.

We now get the patients with get_mrn_to_patient which takes into account
merged MRNs.

We then match with bed status using patient_id as we assume bed status
already takes into account merged MRNs.

Commit:
Renames get_patients_from_mrns to find_patients_from_mrns

Commit:
Reenables the TB appointment load

This change reenables the tb appointment load so that the TB doctors
have up to date information about the state of TB appointments.

Commit:
Disable the clinic day appointment load

We are still weary that this job sometimes hangs. As such Only run the
job at the beginning of each day to minimise the risk.

Commit:
Removes the unused "updated" variable in the ipc status loader

Commit:
Makes the load_ipc_status mgmt command load all patients

Previously it only loaded patients who had a bed status but constraint
was for testing purposes only.

Commit:
Adds a comment to the load ipc bed status command

Adds a comment to the load ipc bed status command to explain why a
patient might not be present in the MRN to patient dictionary

Commit:
Merge branch 'cerner-merge-branch' into merge-get-or-create-patient

Commit:
Merge branch 'merge-get-or-create-patient' of github.com:openhealthcare/elcid-rfh into merge-get-or-create-patient

Commit:
Make the covid calculator tests use tz aware datetimes

Use of timezone unaware datetimes creates warnings, so lets not use them
in tests and cut down on the noise.

Commit:
Reduce the logging in tests done by plugins.tb.tests.test_loader

We do not need to log info warnings when running tests.

Commit:
Changes get_active_mrn_and_merged_mrn_data to deal with exceptions

There are 3 sorts of issue we can have with
get_active_mrn_and_merged_mrn_data.

1. We get multiple rows for a single MRN in the masterfile.
This should never happen. In this instance we raise a ValueError because we
should not proceed in this situation.

2. The patient does not exist in the CernerMasterFile.
This should usually not happen. There are cases where we have patients in our
system that are not registered  in the RFH Cerner, these situations need
to be handled by the calling code.

3. There is a typo in the structure of the merge information.
We have instances where there are multiple active patients and instances
where there are only inactive instances. In this case we should allow
the creation of the patient and send us an email so that we can look
into it.

Commit:
Fixes flake8 formatting issues in the intrahospital_api.loader

Commit:
Adds in the boolean rfh_patient argument to get_or_create_patient

Sometimes we deal with patients who are not RFH patients for example
patients from other hospitals who our doctors are advising on.

In these cases they should be able to create a patient but we should not
look for RFH data such as imaging for them.

This argument tells the calling code to not look in upstream databases
for information about the patient.

Commit:
Reformats merge_patient with black

Commit:
Adds hardcoded include/ignore lists for models in merge_patient

Hard codes a list of patient/episode related models that should be
copied over. Adds an patient/episode ignore lists in the test.

If a related model is not in either of these lists an error is raised by
the unit test. This forces us to explicitly make sure that related
models are considered when they are added.

Commit:
Fixes up function names/docstrings in merge_patient and test_merge_patient

Hopefully this makes it more intuitively readable.

Commit:
Removes now unused functions from merge_patient

Commit:
Renames move_tagging to update_tagging as it does not actually move tags

In merge_patient, rename move_tagging to update_tagging

Commit:
Changes episode_list.html to always show lab tests

Removes the if clause around the view_lab_test_trends role and deletes
the related template

davidmiller committed:
Merge pull request #1367 from openhealthcare/merge-patient

Adds the merge_patient function

davidmiller committed:
Merge pull request #1393 from openhealthcare/reduce-test-warnings

Make the covid calculator tests use tz aware datetimes

davidmiller committed:
Updated contact tracing flow for location history

Commit:
Minor variable spelling fix in plugins.admissions.views

Commit:
Converts tabs to spaces in the admissions.test_views

Commit:
Adds unit tests for the transfer history logic in the LocationHistoryEncounterContactsView

This does not change any logic but adds unit tests that check all the
possibilties as the logic is not obvious (but it works).

fredkingham commented:
A few suggestions so far

Long locations do weird things to the buttons http://192.168.21.203/#/admissions/encounter/129079734/contacts/

You also maybe want to leave transfer reason as blank rather than "None"

This page is not working http://192.168.21.203/#/admissions/encounter/116200754/contacts/ (could be an issue with the data sync?)

davidmiller committed:
Use the same chronological order in location history detail and encouner contacts

davidmiller committed:
Show potential contacts count on encounter contacts page

davidmiller committed:
Merge branch 'contact-tracing-preview' of github.com:openhealthcare/elcid-rfh into contact-tracing-preview

davidmiller committed:
Use abs encounter slice id for toggling slice contacts

Commit:
Changes the version number to v0.107.1

Commit:
Restores the services menu removed in 4bd8e3c9edf206ec69efedd82bb95b419dafc930

An unknown issue was making the services menu not work on prod. The drop
down worked but clicking the individual services did not work. As such
the services menu was removed. This commit restores it.

Commit:
Moves the uib-dropdown-toggle to be on a child element in the service menu

We had an issue that clicking on the dropdown elements in the services
menu did not change the page. This is because the uib-dropdown-toggle
was overriding all links so that they toggled the menu. This resolves
this by moving the dropdown toggle to the <a> which does not encompass
the rest of the menu.

davidmiller committed:
Add a 30 day history view to IPC wards.

davidmiller committed:
Merge branch 'v0.107' into contact-tracing-preview

davidmiller committed:
Fixup, clarity in headings

davidmiller committed:
Merge branch 'contact-tracing-preview' of github.com:openhealthcare/elcid-rfh into contact-tracing-preview

davidmiller committed:
For a single slice, admit/discharge is a misleading heading, use start/end

Commit:
Makes load all transfer history only load in spells = 1 and in trans hist = 1

In trans hist and in spells determine if a transfer history has been
removed upstream.

We also remove an unneeded order by clause.

Commit:
Adds the zero prefixed patient merge management command

Commit:
Reject any attempt to hit the server which are not our allowed hosts servers

Commit:
Ignore complex merges when we merge zero prefixed patients

davidmiller committed:
There's nothing merged about either this query or function

davidmiller committed:
While implementation of parse_merge_comments

davidmiller committed:
Functions claiming to return things should include a return statement

davidmiller committed:
Fixup pro tip: Don't access variables that are created on the following line.

davidmiller committed:
Fixup: append not assign

Commit:
Changes the way that we reload zero prefixed patients

We reload patients who either have zero prefixes in our system or who
have zero prefixes in any of the upstream systems. This is because some
patients in our system preexisted the removal of zero prefixes in the
upstream systems.

Commit:
Fixes the comments in the load all transfer history command

Commit:
Fixes update_demographics.parse_merge_comments to assign the correct active MRN

It was assigning the MRN that was passed in as the active MRN. This
commit changes it so that it assigns the next_mrn variable. It renames
the mrn variable to initial_mrn as we are dealing with a lot of MRNs so
hopefully this makes it easier to read.

Commit:
Fixes unit tests for update_demographics.parse_merge_comments

davidmiller committed:
Merge pull request #1414 from openhealthcare/recursion-adjustments2

Recursion adjustments2

davidmiller committed:
Merge pull request #1361 from openhealthcare/merge-get-or-create-patient

Adds the get_or_create_patient function that looks for patients by MRN

davidmiller committed:
Merge pull request #1374 from openhealthcare/merge-icu-handover

Merge icu handover

davidmiller committed:
Merge pull request #1401 from openhealthcare/contact-tracing-preview-tests

Contact tracing preview tests

davidmiller committed:
Merge pull request #1410 from openhealthcare/load-all-transfer-histories-fix

Makes load all transfer history only load in spells = 1 and in trans hist = 1

davidmiller committed:
Merge pull request #1409 from openhealthcare/1071-with-services-menu

1071 with services menu

Commit:
Merges v0.107.1 into the cerner-merge-branch

davidmiller committed:
Sideroom status: initial models

davidmiller committed:
Merge branch 'v0.107.1' of github.com:openhealthcare/elcid-rfh into contact-tracing-preview

davidmiller committed:
Sideroom list to show extra fields, new contact tracing screen

davidmiller committed:
Merge branch 'contact-tracing-preview' of github.com:openhealthcare/elcid-rfh into contact-tracing-preview

Commit:
Changes the record panel edit buttons to be faded on non-hover

Opal default is that these icons are not there if not hovered. This
change makes them a pale grey if not hovered.

Commit:
Forces the record panel edit icon to be a pointer for the teams panel

At the moment the teams panel does not have the pointer class so its not
a pointer. This fixes that without an opal change or a larger overriding
the teams panel change.

Commit:
Changes the clinical advice and observations to have faded icons

This changes it so that the clinical advice and observations panels have
icons that are always there but become darker when hovered over.

davidmiller committed:
Merge pull request #1416 from openhealthcare/162-shaded-buttons

162 shaded buttons

davidmiller committed:
Previous MRN is not a useful front end field

davidmiller committed:
Link to sideroom list if on a sideroom, change note name

Commit:
Bug fixes clinical_timeline.controller.test.js

Remove "fit" (ie only run this test) from clinical_timeline.controller.test.js

davidmiller committed:
Merge pull request #1418 from openhealthcare/fix-bad-fit

Bug fixes clinical_timeline.controller.test.js

Commit:
Moves IPC clinical advice errors to the bottom of the form

Show the error messages for IPC clinical advice at the bottom of the
form in the same way as we do for the standard infection service
clinical advice and in TB

Commit:
Merge branch 'v0.107.1' into cerner-merge-branch-merge

davidmiller committed:
Merge pull request #1419 from openhealthcare/ipc-error-messages

Moves IPC clinical advice errors to the bottom of the form

davidmiller committed:
Merges in 107.1

davidmiller committed:
Merge pull request #1370 from openhealthcare/quick-discharge-summary-speed-improvement

Changes the discharge summary meds creation to use bulk create

davidmiller committed:
Merge transfer-hsitory into cerner-ergebranch

davidmiller committed:
Merge branch 'cerner-merge-branch' of github.com:openhealthcare/elcid-rfh into cerner-merge-branch

davidmiller committed:
Merge remote-tracking branch 'origin/135-write-to-epr-initials' into cerner-merge-branch

davidmiller committed:
Don't show status badge from a CI we don't use

Commit:
Merge branch 'cerner-merge-branch' into cerner-merge-branch-merge

fredkingham committed:
Merge pull request #1415 from openhealthcare/cerner-merge-branch-merge

Cerner merge branch merge

Commit:
Removes ipc.loader as its not used.

ipc.loader contained the function load_all_extra_ipc_patients is only
there for development and tesing
(https://github.com/openhealthcare/elcid-rfh/pull/1373/files#r1115543430)

davidmiller committed:
IPC fixups:
Sorting of wards in sideroom list.
styling of contacts show hide buttons
show current location in episode contacts

Commit:
Makes the user admin form error on duplicate email

If a user is added with an existing email or if a user is edited with an
existing email it should raise a Validation error.

Commit:
Adds a duplicate email check to create_covid_user

If there already exists a user with that email, create_covid_user now
raises an error rather than create another user with the same email.

Commit:
Fixes create covid user unit tests

Locally we don't run migrationsm, on CI we do so what was passing
locally was failing on CI. This fixes that.

Commit:
Changes the IPC Status load to create patients

The IPC status load should create patients if they don't exist

Commit:
Moves opal-observations into the elcid package

Commit:
Moves obs into plugins.obs

Commit:
Removes the unused obs.admin file

Commit:
Make the Observation model inherit from PreviousMRN

Todo this we need to change the one to one foreign key relationship in
elcid.models to use the string syntax

Commit:
Changes transfer history/encounter batch loads

Changes the encounter and transfer history batch loads to work with
mergedMRNs

Commit:
Changes batch_load2 to user merged MRN lookups

Commit:
Makes the error for multiple active mrns clearer

We were not stating the multiple active MRNs in the multiple active MRN
message. Lets do that to make it clearer.

Commit:
Adds a check management command to make sure our merging is working

Adds a managment command that makes sure...

* We do not have MRNs that are both in Demographics.hospital_number and
  MergedMRN.

* We do not have MRNs that are marked as active and merged upstream that
  do not have any merged MRNs

* We do not have MRNs that are marked as inactive upstream and active in
  our system.

Commit:
Fixes the unit tests for update demographics

Commit:
Fixes unit tests in the obs plugin

Commit:
Changes elcid admin to use spaces not tabs

Commit:
Moves the admin User forms out of forms.py into admin.py

Lets keep admin functionality in admin.py

Commit:
Improves logging in merge_zero_prefixed_patients

Commit:
Changes elcid utils to handle None as an MRN

In the bed status table one of the MRNs is marked as None. We should
handle this without raising an exception as its no different from an
empty string MRN.

Commit:
Makes the bed status loader not create patients with empty MRNs

The bed status loader should not create create patients with empty MRNs
or with empty string

Commit:
Makes the error messages include the MRN in create_rfh_patient_from_hospital_number

create_rfh_patient_from_hospital_number is often called as a large batch
job. Not including the MRNs in the error messages makes it very
difficult to debug, so lets put them in.

Commit:
Adds in a unit test to check an error in the intrahospital_api.loader

Commit:
Bug fixes variable confusion in merge_zero_prefixed_patients

Commit:
Changes the merge zero patients to call the discharge summary command

Fetching all discharge summaries is much quicker than doing it
individually

Commit:
Changes the appointment fields so that they are all lower case

We had a random uppercase in one of the fields. I assume this is a typo
and have resolved. It was causing issues with the csv import.

fredkingham committed:
Update merge_zero_prefixed_patients.py

Commit:
Changes the zero merge to only resync demographics

Changes the merge_zero_prefixed_patients to only reload demographics as
we think this is enough. It
also removes some transactions.

Commit:
Merge branch 'zero-merge-mgmt-command' of github.com:openhealthcare/elcid-rfh into zero-merge-mgmt-command

Commit:
Changes print statements to logger statments

Changes print statements to logger statements in the merge zero prefixes
mgmt command

Commit:
strip leading zeros in demographics.update_from_dict

Commit:
Removes the fetch discharge summaries from the zero prefix mgmt command

The fetch discharge summaries command takes a long time. As we are not
syncing any other data (apart from demographics) as part of this mgmt
command, lets remove fetch discharge summaries.

Commit:
Changes the merge zero command to lookup against integers

Patient ids are integers so change the list of COMPLEX_MERGES ids to
integers

Commit:
Improves the logging in merge_zero_prefixed_patients

Lets include the elcid patient ids in the merge statement for more
transparency

Commit:
Bumps the version from 107.1 to 108

Commit:
Changes a print statement to a logger statement

Commit:
Adds unit tests for creating/merged transfer histories

Make sure that in the batch we should create patients if they do not
exist.

Commit:
Fixes a typo in the admissions loader test

Commit:
White space fix in the admissions.loader

Commit:
Changes the strip zero logic in demographics to the save method

We were using it update_from_dict but its behaviour we never want to
happen so we might as well put it in the save method.

davidmiller committed:
Merge pull request #1413 from openhealthcare/zero-merge-mgmt-command

Zero merge mgmt command

davidmiller committed:
Merge pull request #1426 from openhealthcare/merge-lab-tests

Changes batch_load2 to user merged MRN lookups

Commit:
Makes us not raise an error is a merged MRN in the upstream result

Commit:
Merge branch 'v0.107.1' into v0.108

Commit:
Removes duplicate server_name entry in the nginx conf

Commit:
Copy over lab tests as part of merge patient

We have lab tests in our system that do not exist in the upstream
system so we need to copy lab tests over.

Commit:
Removes the merge_zero_prefixed_patients mgmt cmd

This has now been run on prod and is no longer needed.

Commit:
Removes whitespace change in elcid/forms.py

Removes white space, no functional effect but makes the diff cleaner.

davidmiller committed:
Merge pull request #1411 from openhealthcare/nginx-allowed-hosts

Reject any attempt to hit the server which are not our allowed hosts servers

davidmiller committed:
Merge pull request #1441 from openhealthcare/remove-merge-zero

Removes the merge_zero_prefixed_patients mgmt cmd

davidmiller committed:
Merge pull request #1440 from openhealthcare/merge-lab-tests-1

Copy over lab tests as part of merge patient

davidmiller committed:
Merge remote-tracking branch 'origin/v0.108' into contact-tracing-preview

davidmiller committed:
Show Sex on Siderooms

davidmiller committed:
show newlines for actions

fredkingham committed:
Merge pull request #1444 from openhealthcare/v0.108

Merge 108 into the cerner merge branch

davidmiller committed:
Merge remote-tracking branch 'origin/cerner-merge-branch' into contact-tracing-preview

Commit:
Make sure the tb patient load handles merged MRNs

Commit:
Creates Nationality.birth_place and copies the data from Demographics.birth_place

Commit:
Changes the front end to populate TB.birth_place to be

Commit:
Makes covid create_followup_episodes work with merged mrns

Commit:
Changes the covid loader to work with MergedMRNs

davidmiller committed:
Merge pull request #1423 from openhealthcare/168-disallow-multiple-user-email-addresses

168 disallow multiple user email addresses

davidmiller committed:
Merge pull request #1397 from openhealthcare/remove-flawed-comment-tags

Removes useless comment tags

Commit:
Fixes the bed status loader so bed status.patient can be None

BedStatus is a list of locations, we expect patient to be None frequently.

davidmiller committed:
Merge pull request #1449 from openhealthcare/fix-bed-status

Fixes the bed status loader so bed status.patient can be None

davidmiller committed:
Merge pull request #1445 from openhealthcare/merge-tb-patient-load

Make sure the tb patient load handles merged MRNs

davidmiller committed:
Merge pull request #1447 from openhealthcare/merge-patient-covid-loader

Merge patient covid loader

Commit:
Allow reverse code for copying over birth place

In theory if we are reversing this we are backing out the move to
keeping the data in nationality so copy it back over to demographics

Commit:
Remove the rfh_patient from create_rfh_patient_from_hospital_number

We don't think we need this functionality at the moment we could restore
it in future

davidmiller committed:
Merge pull request #1446 from openhealthcare/tb-birth-place

Tb birth place

davidmiller committed:
Merge pull request #1451 from openhealthcare/remove-get-or-create-rfh-patient-flag

Remove the rfh_patient from create_rfh_patient_from_hospital_number

Commit:
Renames a variable in admissions loader

Renames to_create to encounters_to_create so that it doesn't get
confused with any of the patient creation

Commit:
Merge branch 'cerner-merge-branch' into merge-encounters-and-transfer-history

Commit:
Removes upstream transfer history rows with invalid MRNs

We don't want to save upstream rows where the MRN is an empty string,
only 00s or None

Commit:
Remove upstream encounter rows with invalid MRNs

Filter encounter rows with invalid MRNs

Commit:
Fixes a variable name in the admissions tests

Commit:
Merge branch 'cerner-merge-branch' into merge-encounters-and-transfer-history

Commit:
Fixes a bug where our_merge_datetime was not set.

Fixes a bug where our_merge_datetime was not set when creating new
patients.

Commit:
Renames 'load_demographics' to 'search_upstream_demographics'

This function is only used by the api end point for add patient

Commit:
Look up the merged MRN in search_upstream_demographics

We do the merged MRN lookup in the upstream demographics search as if we
do it later the table does not have access to all the information

Commit:
Removes the  view from the intrahospital_api

Commit:
Removes the rest of the intrahospital_api views

Commit:
Removes run_async=False from the admissions loader

We don't need to create the patients synchronously and none of the other
loaders do.

Commit:
Removes writeback.write_advice_upstream

Removes writeback.write_advice_upstream, the related signal and task

Commit:
Removes intrahospital_api.writeback

Removes intrahospital_api.writeback as we are no longer interested in
writing to that table.

It also removes the cron job, deployment of the cron job and the setting

, the related cron

davidmiller committed:
Merge pull request #1455 from openhealthcare/remove-old-intrahospital_api-views

Remove old intrahospital api views

Commit:
Changes admissions.loader as we don't need to strip MRNs

The MRNs have had empty MRNs removed by the is_valid_mrn line so we
don't need to do it again.

We don't need to run MRN.strip() as there are no MRNs within the
encounter table with preceding zeros.

Commit:
Changes update_demographics merge messages to be warn not error

We have a significant amount of upstream rows that involve flawed
merges. If we error message on each of these we get a significant amount
of emails. This is not an effective way of managing these issues.

Commit:
Removes an unused import in elcid.utils

We are importing Patient but not using it

davidmiller committed:
Merge pull request #1436 from openhealthcare/explicit-error-messages

Explicit error messages

Commit:
If a patient id is passed to a celery task that does not exist, do not error

Commit:
Changes the admissions loader to be clearer about MRN cleaning

Before creating an MRNs we need to remove Nones and empty strings. The
code prior to this should have been doing this but was not.

Commit:
Removes an unused import statements in merge_patient

Commit:
Mocks load_patient in a unit test to stop excessive logging

Mock loader.load_patient in test_merge_patient so that we don't do
unnecessary noisy logging.

Commit:
Makes merge_patient log patients.to_dict

Makes merge_patient log the patients before and after the merge.

davidmiller committed:
Merge pull request #1456 from openhealthcare/remove-write-back

Remove write back

davidmiller committed:
Merge pull request #1457 from openhealthcare/down-grade-merge-problems-logging

Changes update_demographics merge messages to be warn not error

davidmiller committed:
Merge pull request #1461 from openhealthcare/async_celery_load_patient_missing

If a patient id is passed to a celery task that does not exist, do not error

Commit:
Simplifies the check merged patients command.

Simplifies check merge patient so we only check if there are any missing
active patients or any missing inactive patients

davidmiller committed:
Merge pull request #1458 from openhealthcare/remove-unused-import

Removes an unused import in elcid.utils

Commit:
Changes the check_merge command to explicitly ignore flawed rows

Commit:
Improves comments on chech_merged_patients

Commit:
Send a descriptive email rather than logger.error

Also don't raise a ValueError, handle all data issues by sending an
email.

Commit:
Handle flawed active MRNs in the check_merged_patients cmd

There are a lot of active MRNs with no merge comments, these can be
ignored for the purpose of this command.

There a alot of acitve MRNs where the upstream data is flawed. These
should be check on an individual basis for the time being. All current
flawed patients within elcid have been checked and they are an upstream
data issue and not ours.

davidmiller committed:
Merge pull request #1452 from openhealthcare/our_merge_datetime_fix

Fixes a bug where our_merge_datetime was not set.

Commit:
Changes merge patient to log to a seperate file

Outside of standard logging, just while we want to easily make sure that
merge patient works as expected.

Commit:
Fixes the previous MRN message at the bottom of modals

Prior to this it was not showing if the item did not have a created or updated. These are not relevent for previous MRNs.

Commit:
Fixes created/updated/previous MRN modal footer for formItem

fredkingham committed:
Merge pull request #1464 from openhealthcare/v0.108

Merges the latest changes from 108 into the cerner merge branch

Commit:
Adds another active MRN to ignore in check_merged_patients

Commit:
Changes merge_patient to log the old patient.to_json to seperate file

Changes it so that we log the old patient.to_json to a seperate file.

In the log we will log everything i.e. old patient json, new patient json, merge patient json

Commit:
Improves comments in check_merged_patients

Commit:
Adds in a merge_dup_episode_categories mgmt command

A patient should never have multiple episodes with the same episode
category.

This was not always the case. This PR merges the duplicate episode
categories. The logic is mostly taken from merge_patient but does not
populate previousMRN fields

Commit:
Removes settings.ADD_PATIENT_LAB_TESTS as this is False

Prior to the the roll out of lab tests this flag was to allow us to easily
switch to using lab tests. We now will always use lab tests so lets
remove this redudant setting.

Commit:
Changes add patient pathway to use the new create patient API

In the add patient pathway where possible use get_or_create_patient.

We do get_or_create in the off chance that the patient has been created
in the interim between the search step on the pathway and when we press
save.

We still call super to make sure we save the related subrecords.

Commit:
Adds 2 additional checks to check_merged_patients

Adds 2 additional checks to check_merged patients.

* Makes sure that every MRN with a merged MRN in our system is an MRN marked as an active merged mrn upstream.
* Makes sure that every merged MRN in our system is an MRN marked as inactive upstream

davidmiller committed:
Merge pull request #1465 from openhealthcare/previous-mrn-modal-fix-1

Previous mrn modal fix 1

davidmiller committed:
Merge pull request #1425 from openhealthcare/merge-encounters-and-transfer-history

Changes transfer history/encounter batch loads

davidmiller committed:
Merge pull request #1462 from openhealthcare/log-before-and-after-merge

Log before and after merge

davidmiller committed:
Allow ScrollTo Ward; exclude known junk wards

Commit:
Moves the check_merged_patient into monitoring

Moves the check_merged_patient mgmt command into monitoring.

Commit:
Removes a layer of indirection in merge_dup_episode_categories

Commit:
Renames variables in merge_dup_episode_categories

references to old_parent/new_parent aren't helpful when the parent is
always and episode, lets call a spade a spade.

Commit:
Changes merge_dup_episodes logging

Rather than logging patient id and episode id, log the full episode of
the old episode

Commit:
Changes update_tagging to require keyword arguments

Commit:
Improves the comment on merge_patient.update_tagging

Commit:
Adds a logger for the ipc module

Commit:
Changes load_ipc_status to use the logger module

Adds some additional logging to show the status of the load

davidmiller committed:
Merge pull request #1469 from openhealthcare/ipc-logging-1

Adds IPC logging

davidmiller committed:
Merge pull request #1468 from openhealthcare/merge-tagging

Updates the api call and comments on update tagging

Commit:
Removes the RNOH logic from the add patient pathway.

Removes the RNOH logic from the add patient pathway as although we do
want it, we don't want it now.

Commit:
Changes get_or_create_patient to handle CernerPatientNotFoundException

If the patient is not found upstream then get_or_create_patient should
create the patient anyway.

Commit:
Changes the AddPatientPathway as it no longer needs to check for CernerPatientNotFoundException

We changed get_or_create_patient so that it does not throw
CernerPatientNotFoundException, so take this check out of the
AddPatientPathway

Commit:
Removes unused imports in elcid/pathways.py

davidmiller committed:
Update check_merged_patients.py

Commit:
Merge branch 'v0.108' into cerner-merge-branch

Commit:
Renames the elcid.time_logger to elcid.utils

Renames the logger as we will want to use it for time logging but also
logging in other utility functions

Commit:
Adds the send_email function to elcid.utils

Adds the send_email function with sends an email to the admins prefixing
the subject with the brand name.

It also logs before and after to help track back if there are any issues
with the hospital email server.

Commit:
Changes batch_load2 to use the new send_email util

Commit:
Changes check_disk_space to use utils.send_email

Commit:
Changes the feed_alert_email to use utils.send_email

davidmiller committed:
Merge pull request #1470 from openhealthcare/send-email-util

Send email util

Commit:
Merge branch 'v0.108' into cerner-merge-branch

davidmiller committed:
Merge pull request #1466 from openhealthcare/merge-dup-episode-categories

Adds in a merge_dup_episode_categories mgmt command

davidmiller committed:
Fixup: Cerner is a noun

davidmiller committed:
Fixup

Docstring capitalisation, typos, clarity

davidmiller committed:
Fixup: Capitalization

davidmiller committed:
Merge pull request #1454 from openhealthcare/new-patient-demographics-search-with-merged-mrn

New patient demographics search with merged mrn

Commit:
Merge branch 'cerner-merge-branch' into chech-merge-patients

Commit:
Use utils.send_email in check_merged_patients

We now have a new utility for sending emails to admins so lets use this rather
than duplicate code.

Commit:
Merge branch 'chech-merge-patients' of github.com:openhealthcare/elcid-rfh into chech-merge-patients

Commit:
Fixes a unit test in intrahospital_api.test.test_loader

No functional change, there was a capitalisation fix that changed a
message. This fixes the related unit test.

Commit:
Merge branch 'cerner-merge-branch' into chech-merge-patients

davidmiller committed:
Merge pull request #1428 from openhealthcare/chech-merge-patients

Adds a check management command to make sure our merging is working

davidmiller committed:
Merge pull request #1491 from openhealthcare/cerner-merge-branch

Cerner merge branch

davidmiller committed:
Improve Ward nav styles

davidmiller committed:
Allow filtering of the sideroom list by organism, and show counts of sex

davidmiller committed:
Bugfix: Closed bays should only be displayed on the isolation list for that hospital

davidmiller committed:
Highlight what is a sideroom / temp sideroom

davidmiller committed:
merge upstream contact-tracing-preview

davidmiller committed:
Remove editable IPC reason for interaction

davidmiller committed:
Link to sideroom list from detail page when a closed open bay

davidmiller committed:
Improve display of wards for RFH

davidmiller committed:
Trial of Per flag lab numbers

davidmiller committed:
Add lab detail modal and directive to render links into text

davidmiller committed:
Expiry is the other way round

davidmiller committed:
Sometimes a patient doesn't have covid, then gets it. we shouldn't be unable to handle that.

davidmiller committed:
Make sideroom status editable from the sideroom list
